### ISSUE TITLE
Configurable correlation id

### ISSUE THREAD

**Issue by skjolber** [zalando/logbook#311]
Make logging the correlation id optional. 

## Context
We add a unique request-id to the MDC context for each incoming request. This allows us to see all log statements resulting from a call. 

If interesting, we also add a hash of the authorization and our own trace id. 

## Possible Implementation
Boolean configuration value

## Your Environment
Kubernetes / Stackdriver - single log file.

**Comment by whiskeysierra** [comment]
What exactly is your intent? Sounds a lot like https://github.com/zalando/tracer to me.

**Comment by skjolber** [comment]
Basically Logbook uses '[correlation](https://github.com/zalando/logbook/blob/e81f30410dcb37a838bbb90c6ee646d6a43bed14/logbook-core/src/main/java/org/zalando/logbook/JsonHttpLogFormatter.java#L93)' to connect a request to a response log statement. We a making log statements inbetween the request and response (on the same server, in the same log output bucket) and so now add 'our own correlation' to the root of the JSON log statement. 

So in other words I'd like the 'correlation' Logbook is adding to not be a child in the request/response logging JSON tree, but rather added to the MDC so that it is present in the root of the logged JSON tree and thus present on all log statements between Logbook request and response filters.

**Comment by whiskeysierra** [comment]
The correlation feature in logbook is purely to connect request and response. The primary use case is described in [Logbook + Scalyr](https://github.com/zalando/logbook/blob/master/docs/scalyr.md).

What you describe is almost to the nail what Tracer offers. Those are two different aspects, hence why we split them up into different projects.
